src/sentry/rules/actions/utils.py

```diff
@@ -62,26 +79,33 @@ def get_frequency_label(value_str: str | None) -> str | None:
     return None
 
 
+def convert_data(data: list[dict[str, Any]]) -> DefaultDict[str, dict[str, str | int]]:
+    converted_data: DefaultDict[str, dict[str, str | int]] = defaultdict(dict)
+    for datum in data:
+        converted_data[datum.get("id")] = datum
```

Comment:
Just to note:

IF there's ever data with duplicated `id`s, the latest one will overwrite all the others.

This may be an unexpected edge case limitation.
Otherwise - we could potentially hash the `id` with whatever other attrs are available to us to construct some other identifier 🤷 

but just might be worth noting this limitation for posterity